src/security/tests/cli_security_tests.c

```diff
@@ -348,6 +351,388 @@ test_request_credentials_returns_raw_bytes(void **state)
+						    ARRAY_SIZE(bad_gids), &perms),
+			-DER_NONEXIST);
+
+	D_PRINT("= no owner in prop\n");
+	prop_no_owner = get_pool_acl_prop(NULL, "group@", acl);
+	assert_rc_equal(dc_sec_get_pool_permissions(prop_no_owner, uid, gid, NULL, 0,
+						    &perms),
+			-DER_INVAL);
+	daos_prop_free(prop_no_owner);
+
+	D_PRINT("= no owner-group in prop\n");
+	prop_no_group = get_pool_acl_prop("user@", NULL, acl);
+	assert_rc_equal(dc_sec_get_pool_permissions(prop_no_group, uid, gid, NULL, 0,
+						    &perms),
+			-DER_INVAL);
+	daos_prop_free(prop_no_group);
+
+	D_PRINT("= no ACL in prop\n");
+	prop_no_acl = get_pool_acl_prop("user@", "group@", NULL);
+	assert_rc_equal(dc_sec_get_pool_permissions(prop_no_acl, uid, gid, NULL, 0,
+						    &perms),
+			-DER_INVAL);
+	daos_prop_free(prop_no_acl);
+
+	daos_prop_free(pool_prop);
+	daos_acl_free(acl);
+}
+
+static void
+test_get_cont_perms_invalid_input(void **state)
+{
+	daos_prop_t	*prop_no_owner;
+	daos_prop_t	*prop_no_group;
+	daos_prop_t	*prop_no_acl;
+	daos_prop_t	*cont_prop;
+	struct daos_acl	*acl;
+	uid_t		uid = geteuid();
+	gid_t		gid = getegid();
+	gid_t		bad_gids[2] = {getegid(), (gid_t)-1};
+	uint64_t	perms;
+
+	acl = daos_acl_create(NULL, 0);
+	assert_non_null(acl);
+
+	cont_prop = get_cont_acl_prop("user@", "group@", acl);
+
+	D_PRINT("= NULL cont prop\n");
+	assert_rc_equal(dc_sec_get_cont_permissions(NULL, uid, gid, NULL, 0, &perms),
+			-DER_INVAL);
+
+	D_PRINT("= NULL perms param\n");
+	assert_rc_equal(dc_sec_get_cont_permissions(cont_prop, uid, gid, NULL, 0, NULL),
+			-DER_INVAL);
+
+	D_PRINT("= NULL gids with num gids > 0\n");
+	assert_rc_equal(dc_sec_get_cont_permissions(cont_prop, uid, gid, NULL, 1, NULL),
+			-DER_INVAL);
+
+	D_PRINT("= bad uid\n");
+	assert_rc_equal(dc_sec_get_cont_permissions(cont_prop, (uid_t)-1, gid, NULL, 0,
+						    &perms),
+			-DER_NONEXIST);
+
+	D_PRINT("= bad gid\n");
+	assert_rc_equal(dc_sec_get_cont_permissions(cont_prop, uid, (gid_t)-1, NULL, 0,
+						    &perms),
+			-DER_NONEXIST);
+
+	D_PRINT("= bad gid in list\n");
+	assert_rc_equal(dc_sec_get_cont_permissions(cont_prop, uid, gid, bad_gids,
+						    ARRAY_SIZE(bad_gids), &perms),
+			-DER_NONEXIST);
+
+	D_PRINT("= no owner in prop\n");
+	prop_no_owner = get_cont_acl_prop(NULL, "group@", acl);
+	assert_rc_equal(dc_sec_get_cont_permissions(prop_no_owner, uid, gid, NULL, 0,
+						    &perms),
+			-DER_INVAL);
+	daos_prop_free(prop_no_owner);
+
+	D_PRINT("= no owner-group in prop\n");
+	prop_no_group = get_cont_acl_prop("user@", NULL, acl);
+	assert_rc_equal(dc_sec_get_cont_permissions(prop_no_group, uid, gid, NULL, 0,
+						    &perms),
+			-DER_INVAL);
+	daos_prop_free(prop_no_group);
+
+	D_PRINT("= no ACL in prop\n");
+	prop_no_acl = get_cont_acl_prop("user@", "group@", NULL);
+	assert_rc_equal(dc_sec_get_cont_permissions(prop_no_acl, uid, gid, NULL, 0,
+						    &perms),
+			-DER_INVAL);
+	daos_prop_free(prop_no_acl);
+
+	daos_prop_free(cont_prop);
+	daos_acl_free(acl);
+}
+
+static void
```

Comment:
Just confirming, this is the same function as in the daos_test daos_container.c source file but I assume needs to be copy/pasted in each separate test file? Not a big deal at all IMO, just making sure I have it right.